src/test/java/io/lettuce/core/commands/KeyCommandIntegrationTests.java

```diff
@@ -192,6 +194,49 @@ void expiretime() {
         assertThat(redis.expiretime(key)).isEqualTo(expiration.getTime() / 1000);
     }
 
+    @Test
+    @EnabledOnCommand("HEXPIRE")
+    void hexpire() {
+        assertThat(redis.hset(MY_KEY, MY_FIELD, MY_VALUE)).isTrue();
+        // the below settings are required until the solution is able to support listpack entries
+        // see TODOs in https://github.com/redis/redis/pull/13172 for more details
+        assertThat(redis.configSet("hash-max-listpack-entries","0")).isEqualTo("OK");
+        assertThat(redis.configSet("set-max-listpack-value","0")).isEqualTo("OK");
+
+        assertThat(redis.hexpire(MY_KEY, 1, Collections.singletonList(MY_FIELD))).isTrue();
+
+        await().until(() -> redis.hget(MY_KEY, MY_FIELD) == null);
+    }
+
+    @Test
+    @EnabledOnCommand("HEXPIRE")
+    void hexpireExpireArgs() {
+        assertThat(redis.hset(MY_KEY, MY_FIELD, MY_VALUE)).isTrue();
+        // the below settings are required until the solution is able to support listpack entries
+        // see TODOs in https://github.com/redis/redis/pull/13172 for more details
+        assertThat(redis.configSet("hash-max-listpack-entries","0")).isEqualTo("OK");
+        assertThat(redis.configSet("set-max-listpack-value","0")).isEqualTo("OK");
+
+        assertThat(redis.hexpire(MY_KEY,
+                Duration.ofSeconds(1),
+                ExpireArgs.Builder.nx(),
+                Collections.singletonList(MY_FIELD))).isTrue();
+        assertThat(redis.hexpire(MY_KEY,
+                Duration.ofSeconds(1),
+                ExpireArgs.Builder.xx(),
+                Collections.singletonList(MY_FIELD))).isTrue();
+        assertThat(redis.hexpire(MY_KEY,
+                Duration.ofSeconds(10),
+                ExpireArgs.Builder.gt(),
+                Collections.singletonList(MY_FIELD))).isTrue();
+        assertThat(redis.hexpire(MY_KEY,
+                Duration.ofSeconds(1),
+                ExpireArgs.Builder.lt(),
+                Collections.singletonList(MY_FIELD))).isTrue();
+
+        await().until(() -> redis.hget(MY_KEY, MY_FIELD) == null);
```

Comment:
Isn't there a HTTL command yet to query the field TTL?

Reply:
Not implemented yet, so can't use it in the test.
Also I wanted to verify the expiration mechanism and have another test verify the TTL command.

But we can test them both in one place I guess. I have no strong opinion on this matter.